lib/user-interface/react/src/components/chatbot/DocumentSummarizationModal.tsx

```diff
@@ -0,0 +1,270 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import {
+    Modal,
+    Box,
+    SpaceBetween,
+    Button,
+    TextContent,
+    FileUpload, Autosuggest, Textarea, Grid,
+} from '@cloudscape-design/components';
+import { FileTypes, LisaChatSession } from '../types';
+import { useMemo, useState } from 'react';
+import { useAppDispatch } from '../../config/store';
+import { useNotificationService } from '../../shared/util/hooks';
+import { useGetAllModelsQuery } from '../../shared/reducers/model-management.reducer';
+import { IModel, ModelStatus, ModelType } from '../../shared/model/model-management.model';
+import { handleUpload } from './FileUploadModals';
+import { IChatConfiguration } from '../../shared/model/chat.configurations.model';
+import { v4 as uuidv4 } from 'uuid';
+import FormField from '@cloudscape-design/components/form-field';
+
+export type DocumentSummarizationModalProps = {
+    showDocumentSummarizationModal: boolean;
+    setShowDocumentSummarizationModal: (state: boolean) => void;
+    fileContext: string;
+    setFileContext: (state: string) => void;
+    setUserPrompt: (state: string) => void;
+    userPrompt: string;
+    selectedModel: IModel;
+    setSelectedModel: (state: IModel ) => void;
+    chatConfiguration: IChatConfiguration;
+    setChatConfiguration: (state: IChatConfiguration) => void;
+    setInternalSessionId: (state: string ) => void;
+    setSession: (state: LisaChatSession) => void;
+    userName: string;
+    handleSendGenerateRequest: any;
+};
+
+export function DocumentSummarizationModal ({
+    showDocumentSummarizationModal,
+    setShowDocumentSummarizationModal,
+    setFileContext,
+    setUserPrompt,
+    userPrompt,
+    selectedModel,
+    setSelectedModel,
+    chatConfiguration,
+    setChatConfiguration,
+    setInternalSessionId,
+    setSession,
+    userName,
+    handleSendGenerateRequest
+}: DocumentSummarizationModalProps) {
+    const [selectedFiles, setSelectedFiles] = useState<File[] | undefined>([]);
```

Comment:
Pretty sure `useState` by default adds `| undefined` so it isn't necessary. Plus it doesn't look like it is ever set to anything other than an empty array anyways.